### DISCUSSION TITLE
Can someone introduce what’s the purpose of this project? I do not understand it.

### DISCUSSION THREAD

**Discussion by alanjustalan** [acheong08/EdgeGPT#153] [Q&A]
It seems that there is no specific function introduction, which makes me a little confused. Can someone introduce it?

**Comment by EsqerYasen** [comment]
i am using for automation. like i have so many question  only one parameter are different . this project is very usefull.

**Comment by scmanjarrez** [comment]
I use it on my telegram bot, instead of using Microsoft Edge I just use the bot to interact with BingChat (EdgeGPT)

**Reply by wanlinwang** [reply]
Hello, please could you show some codes on how to keep the user's conversation session? In README.md I didn't find one.

**Reply by KyoshiHikari** [reply]
Easiest way to keep the conversation session is to make the "bot" variable global.

**Reply by scmanjarrez** [reply]
> Hello, please could you show some codes on how to keep the user's conversation session? In README.md I didn't find one.

Ups sorry, didn't get a notification from this question. Yes, as kiyoshi said, you can keep ChatBot variable global, or store in in a dictionary to have multiple conversations at the same time but continue the conversation based on the dictionary key. Look here https://github.com/scmanjarrez/EdgeGPT-Telegram-Bot/blob/6ac92394f6d71ccad60d31fedd9ba99f1321b742/src/backend.py#L69, I use a dictionary to save the conversation from a given user (chat_id aka cid) and conversation id (cur_conv) that is stored in the variable ChatBot.chat_hub.request.conversation_id
